products-service/src/main/java/com/petclinic/products/presentationlayer/products/ProductController.java

```diff
@@ -85,5 +89,13 @@ public Mono<ResponseEntity<Void>> changeProductQuantity(@PathVariable String pro
                 .flatMap(request -> productService.changeProductQuantity(productId, request.getProductQuantity()))
                 .then(Mono.just(ResponseEntity.noContent().build()));
     }
+    @GetMapping("/filter")
+    public List<ProductResponseModel> getProductsByType(@RequestParam ProductType productType) {
+        List<Product> products = productService.getProductsByType(productType);
+        // Convert the entities to response models (if needed)
+        return products.stream()
```

Comment:
Instead of using products.stream, it's best to use a Reactive Flux in Repository and Service.